pgrx/src/memcx.rs

```diff
@@ -36,18 +36,18 @@ impl<'mcx> MemCx<'mcx> {
 
     /// Allocate a raw byte buffer `size` bytes in length
     /// and returns a pointer to the new allocation.
-    pub fn alloc_bytes(&self, size: usize) -> NonNull<u8> {
-        let ptr = unsafe { pg_sys::MemoryContextAlloc(self.ptr.as_ptr(), size).cast() };
-        // SAFETY: `pg_sys::MemoryContextAlloc` is not allowed to return NULL, it must error.
-        unsafe { NonNull::new_unchecked(ptr) }
+    pub fn alloc_bytes(&self, size: usize) -> Result<NonNull<u8>, OutOfMemory> {
```

Comment:
I just changed the signature on this one because breaking users of this "direct allocation" API is something I already did once this cycle: it's now `NonNull<u8>`, not `*mut u8`. Also, they probably want to be able to make decisions about error-handling-or-not themselves since they're allocating raw buffers!